File: qwen2.5-coder/sistemaDeTiposRevisado/tipos.js
function calcularPromedioMayoresAUmbral(arreglo, umbral) {
  let suma = 0;
  let contador = 0;

  for (let i = 0; i < arreglo.length; i++) {
    // Convertir a número si es posible
    const valorComoNumero = parseFloat(arreglo[i]);
    
    if (!isNaN(valorComoNumero)) {
      if (valorComoNumero > umbral) {
        suma += valorComoNumero;
        contador++;
      }
    } else {
      console.warn(`El elemento ${arreglo[i]} no es un número válido`);
    }
  }

  // Calcular el promedio
  const promedio = contador > 0 ? suma / contador : 0;

  return promedio;
}

// Ejemplo de uso:
const miArreglo = [10, '20', 'casa', 30.0, '40', 50];
const umbral = 25;
console.log(calcularPromedioMayoresAUmbral(miArreglo, umbral)); // Output: 30

File: qwen2.5-coder/sistemaDeTiposRevisado/tipos.py
def calcular_promedio_mayores_a_umbral(arreglo, umbral):
    # Convertir todos los elementos a floats para realizar la comparación y el cálculo
    numeros = [float(elemento) for elemento in arreglo if isinstance(elemento, (int, float))]
    
    # Filtrar los valores mayores al umbral
    valores_mayores = [numero for numero in numeros if numero > umbral]
    
    # Calcular el promedio si hay valores mayores al umbral
    if len(valores_mayores) > 0:
        promedio = sum(valores_mayores) / len(valores_mayores)
        return promedio
    else:
        return None

# Ejemplo de uso
arreglo = [1, '2', 30.0, '40', 'cinco', 50]
umbral = 25
promedio = calcular_promedio_mayores_a_umbral(arreglo, umbral)
print("Promedio:", promedio) 